tests/e2e/kafka_prod_consum_test.go

```diff
@@ -146,3 +148,128 @@ func requireInternalProducingConsumingMessage(kubectlOptions *k8s.KubectlOptions
 		Expect(consumedMessage).Should(ContainSubstring(currentTime.String()))
 	})
 }
+
+// requireExternalProducerConsumer deploys a kafkaTopic into the K8s cluster
+// and produces, consumes messages and makes comparison between the produced and consumed messages.
+func requireExternalProducerConsumer(kubectlOptions *k8s.KubectlOptions) {
+	When("Externally produce and consume message to Kafka cluster", Ordered, func() {
+		requireDeployingKafkaTopic(kubectlOptions, testTopicName)
+		// requireAvailableExternalKafkaAddress is not necessary now but I think later can be useful
+		// when we need to wait before consuming producing to get LB service ip address and domain resolution
+		// in those cases when externalListener added dynamically.
+		// Currently we dont know much about our testing environment.
+		requireAvailableExternalKafkaAddress(kubectlOptions, "", kafkaClusterName)
+		requireExternalProducingConsumingMessage(kubectlOptions, testTopicName)
+		requireDeleteKafkaTopic(kubectlOptions, testTopicName)
+	})
+}
+
+// requireExternalProducingConsumingMessage gets the Kafka cluster external addresses from the kafkaCluster CR status
+// when externalAddresses is not specified and producing, consuming messages and make comparison between them.
+func requireExternalProducingConsumingMessage(kubectlOptions *k8s.KubectlOptions, topicName string, externalAddresses ...string) {
+	It("Producing and consuming messages", func() {
+		if len(externalAddresses) == 0 {
+			externalAddresses = getExternalListenerAddresses(kubectlOptions, "", kafkaClusterName)
+		}
+		Expect(externalAddresses).ShouldNot(BeEmpty())
+
+		message := time.Now().String()
+		requireProducingSyncMessages(externalAddresses, topicName, message)
+
+		consumedMessages := requireConsumingMessages(externalAddresses, topicName)
+		// This check may fail when the topic is re-created and has some undeleted content.
+		Expect(consumedMessages).Should(HaveLen(1))
```

Comment:
Do we need this check in this case? We might end up reusing stuff in the end. Instead we could just compare with the last message from what we consumed here I think.

Reply:
Will we actually run into this case that " the topic is re-created and has some undeleted content" in the test environment? If not, I think it's better to remove that comment to avoid confusions

Reply:
Ok I will remove it.

Reply:
fixed: [0e03f84](https://github.com/banzaicloud/koperator/pull/991/commits/0e03f843cc92a38634d4adb4b1c22330b51cd011)